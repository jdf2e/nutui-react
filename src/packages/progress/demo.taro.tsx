import React, { useState } from 'react'
import Taro from '@tarojs/taro'
import { Checked, Tips } from '@nutui/icons-react-taro'
import { useTranslate } from '@/sites/assets/locale/taro'
import {
  Cell,
  Progress,
  Button,
  Toast,
  Image,
} from '@/packages/nutui.react.taro'
import Header from '@/sites/components/header'

interface T {
  basic: string
  customStyle: string
  noShowPercentage: string
  customContent: string
  customSize: string
  statusDisplay: string
  dynamicChange: string
  reduce: string
  add: string
}

const ProgressDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      basic: '基础用法',
      customStyle: '设置颜色与宽度',
      noShowPercentage: '显示百分比',
      customContent: '自定义显示内容',
      customSize: '自定义尺寸',
      statusDisplay: '状态显示',
      dynamicChange: '动态改变',
      reduce: '减少',
      add: '增加',
    },
    'zh-TW': {
      basic: '基礎用法',
      customStyle: '設置顏色與寛度',
      noShowPercentage: '顯示百分比',
      customContent: '自定義顯示內容',
      customSize: '自定義尺寸',
      statusDisplay: '狀態顯示',
      dynamicChange: '動態改變',
      reduce: '減少',
      add: '增加',
    },
    'en-US': {
      basic: 'Basic Usage',
      customStyle: 'Custom Style',
      noShowPercentage: 'Show Percentage',
      customContent: 'Custom Content',
      customSize: 'Custom Size',
      statusDisplay: 'Status Display',
      dynamicChange: 'Dynamic Change',
      reduce: 'reduce',
      add: 'add',
    },
  })

  const [value, setValue] = useState(0)
  const [show, SetShow] = useState(false)
  const [toastMsg, SetToastMsg] = useState('')
  const toastShow = (msg: any) => {
    SetToastMsg(msg)
    SetShow(true)
  }
  return (
    <>
      <Header />
      <div className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''}`}>
        <h2>{translated.basic}</h2>
        <Cell>
          <Progress percent={30} />
        </Cell>
        <h2>{translated.customStyle}</h2>
        <Cell>
          <Progress
            percent={30}
            color="var(--nutui-color-primary)"
            background="var(--nutui-brand-1)"
            strokeWidth="15"
          />
        </Cell>
        <h2>{translated.noShowPercentage}</h2>
        <Cell>
          <Progress percent={50} showText />
        </Cell>
        <h2>{translated.customContent}</h2>
        <Cell>
          <Progress percent={60} showText>
            <Image
              width="30px"
              height="30px"
              src="https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png"
            />
          </Progress>
        </Cell>
        <h2>{translated.customSize}</h2>
        <Cell>
          <Progress percent={30} strokeWidth="5" showText />
        </Cell>
        <Cell>
          <Progress percent={50} strokeWidth="10" showText />
        </Cell>
        <Cell>
          <Progress percent={70} strokeWidth="15" showText />
        </Cell>
        <h2>{translated.statusDisplay}</h2>
        <Cell>
          <Progress
            percent={30}
            color="linear-gradient(270deg, rgba(18,126,255,1) 0%,rgba(32,147,255,1) 32.815625%,rgba(13,242,204,1) 100%)"
            animated
          />
        </Cell>
        <Cell align="center">
          <Progress percent={100} />
          <Checked color="green" style={{ margin: '0 5px' }} />
        </Cell>
        <Cell align="center">
          <Progress
            percent={100}
            color="linear-gradient(90deg, rgba(180,236,81,1) 0%,rgba(66,147,33,1) 100%)"
            strokeWidth="15"
          />
          <Tips color="red" style={{ margin: '0 5px' }} />
        </Cell>
        <h2>{translated.dynamicChange}</h2>
        <Cell.Group>
          <Cell align="center">
            <Progress percent={value} />
            <span style={{ margin: '0 5px' }}>{value}%</span>
          </Cell>
          <Cell align="center">
            <Button
              type="default"
              size="small"
              style={{ margin: 8 }}
              onClick={() => {
                if (value <= 0) {
                  toastShow('进度已为0')
                }
                setValue(Math.max(0, value - 10))
              }}
            >
              {translated.reduce}
            </Button>
            <Button
              type="primary"
              size="small"
              style={{ margin: 8 }}
              onClick={() => {
                if (value >= 100) {
                  toastShow('进度已为100%')
                }
                setValue(Math.min(100, value + 10))
              }}
            >
              {translated.add}
            </Button>
          </Cell>
        </Cell.Group>
        <Toast
          type="text"
          visible={show}
          msg={toastMsg}
          onClose={() => {
            SetShow(false)
          }}
        />
      </div>
    </>
  )
}

export default ProgressDemo
