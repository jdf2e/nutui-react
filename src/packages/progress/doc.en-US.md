# Progress

## Intro

Used to show the current progress of the operation.

## Install

```js
// react
import { Progress } from '@nutui/nutui-react';
```

## Demo

### Basic Usage

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Progress percent={30} />
    </Cell>
  );
};
export default App;
```
:::

### Custom Style

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Progress
        percent={30}
        color="rgba(250,44,25,0.47)"
        strokeWidth="20"
        textColor="red"
      />
    </Cell>
  );
};
export default App;
```
:::

### Don't Show Percentage

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Progress percent={30} showText={false} />
    </Cell>
  );
};
export default App;
```
:::

### Text Inside

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Progress percent={60} textInside />
    </Cell>
  );
};
export default App;
```
:::

### Custom Content

:::demo
```jsx
import  React from "react";
import { Progress, Image, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
    <Cell>
      <Progress
        percent={60}
        textInside
      >
        <Image
          width="30px"
          height="30px"
          src="https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png"
        />
      </Progress>
    </Cell>
    </>
  );
};
export default App;
```
:::

## Custom Size

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
    <Cell>
      <Progress
        percent={30}
        size="small"
        textInside
      />
    </Cell>
    <Cell>
      <Progress
        percent={50}
        size="base"
        textInside
      />
    </Cell>
    <Cell>
      <Progress
        percent={70}
        size="large"
        textInside
      />
    </Cell>
    </>
  );
};
export default App;
```
:::

### Status Display

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';
import { Issue } from '@nutui/icons-react';

const App = () => {
  return (
    <>
    <Cell>
      <Progress
        percent={30}
        color="linear-gradient(270deg, rgba(18,126,255,1) 0%,rgba(32,147,255,1) 32.815625%,rgba(13,242,204,1) 100%)"
        status
      />
    </Cell>
    <Cell>
      <Progress percent={100}  textType="icon" />
    </Cell>
    <Cell>
      <Progress
        percent={100}
        color="linear-gradient(90deg, rgba(180,236,81,1) 0%,rgba(66,147,33,1) 100%)"
        strokeWidth="15"
        textType="icon"
        icon={<Issue color="red" />}
      />
    </Cell>
    </>
  );
};
export default App;
```
:::

### Dynamic Change

:::demo
```jsx
import  React, {useState} from "react";
import { Progress, Cell, Button } from '@nutui/nutui-react';

const App = () => {
  const [value, setValue] = useState(0);
  return (
    <>
    <Cell>
      <Progress percent={value} />
    </Cell>
    <Cell>
      <Button
        type="default"
        style={{ margin: 8 }} 
        onClick={() => {
          let num = value;
          if (value <= 0) {
            return false;
          }
          num -= 10;
          setValue(num);
        }}
      >
        减少
      </Button>
      <Button 
        type="primary" 
        onClick={() => {
          let num = value;
          if (value >= 100) {
            return false;
          }
          num += 10;
          setValue(num);
        }}
      >
        增加
      </Button>
    </Cell>
    </>
  );
};
export default App;
```
:::

## Progress

### Props

| Property | Description | Type | Default
|----- | ----- | ----- | -----
| percent | percent | `number` | `0`
| fillColor | Progress bar background color | `string` | `#f3f3f3`
| color | Stroke color | `string` | `linear-gradient(135deg, #fa2c19 0%, #fa6419 100%)`
| strokeWidth | Stroke width | `string` | -
| size | Progress bar and text size, eg `small` `base` `large` | `string` | `base`
| showText | Whether to show text | `boolean` | `true`
| textInside | Progress bar text display position(`false`外显，`true`内显) | `boolean` | `false`
| textColor | Progress bar text color setting | `string` | 外显`#333` 内显`#fff`
| textWidth | Progress bar text width setting | `string` | `35px`
| textBackground | Progress bar text background color setting | `string` | 同进度条颜色
| textType | Progress bar text type setting，`text`(展示文字)/`icon`(展示icon标签) | `string` | `text`
| status | The current state of the progress bar, `true`展示动画效果 | `boolean` | `false`
| icon | Custom Icon | `ReactNode` | `<Checked width={16} height={16} color="#439422"/>`


## Theming

### CSS Variables

The component provides the following CSS variables, which can be used to customize styles. Please refer to [ConfigProvider component](#/en-US/component/configprovider).

| Name | Default Value |
| --- | --- |
| --nutui-progress-border-radius | `12px` |
| --nutui-progress-text-padding | `0 5px` |
| --nutui-progress-text-font-size | `13px` |
| --nutui-progress-text-color | `$title-color` |
| --nutui-progress-text-inner-color | `$gray6` |
| --nutui-progress-insidetext-position-top | `-4px`|
| --nutui-progress-insidetext-position-bottom | `-4px`|
| --nutui-progress-insidetext-border-radius | `5px` |
| --nutui-progress-insidetext-background | `$progress-inner-background-color` |
| --nutui-progress-outer-background-color | `#f3f3f3` |
| --nutui-progress-outer-border-radius | `12px` |
| --nutui-progress-insidetext-border-radius | `5px` |
| --nutui-progress-insidetext-padding | `3px 5px 3px 6px` |
| --nutui-progress-small-font-size | `7px` |
| --nutui-progress-small-padding | `0 4px` |
| --nutui-progress-small-height | `5px` |
| --nutui-progress-small-text-font-size | `7px` |
| --nutui-progress-small-text-line-height | `10px` |
| --nutui-progress-small-text-padding | `2px 4px` |
| --nutui-progress-base-font-size | `9px` |
| --nutui-progress-base-padding | `0 5px` |
| --nutui-progress-base-height | `10px` |
| --nutui-progress-base-text-font-size | `9px` |
| --nutui-progress-base-text-line-height | `13px` |
| --nutui-progress-base-text-padding | `$progress-insidetext-padding` |
| --nutui-progress-base-text-top | `$progress-insidetext-top` |
| --nutui-progress-large-font-size | `13px` |
| --nutui-progress-large-padding | `0 5px` |
| --nutui-progress-large-height | `15px` |
| --nutui-progress-large-text-font-size | `13px` |
| --nutui-progress-large-text-line-height | `18px` |
| --nutui-progress-large-text-padding | `$progress-insidetext-padding` |
| --nutui-progress-large-text-top | `$progress-insidetext-top` |
