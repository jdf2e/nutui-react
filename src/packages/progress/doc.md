# Progress 进度条

## 介绍

展示操作或任务的当前进度。

## 安装

```js
// react
import { Progress } from '@nutui/nutui-react';
```

## 代码演示

### 基础用法

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Progress percent={30} />
    </Cell>
  );
};
export default App;
```
:::

### 设置颜色与宽度

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Progress
        percent={30}
        color="rgba(250,44,25,0.47)"
        strokeWidth="20"
      />
    </Cell>
  );
};
export default App;
```
:::

### 显示百分比

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Progress percent={50} showText />
    </Cell>
  );
};
export default App;
```
:::

### 自定义显示内容

:::demo
```jsx
import  React from "react";
import { Progress, Image, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <Cell>
      <Progress percent={60}>
        <Image
          width="30px"
          height="30px"
          src="https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png"
        />
      </Progress>
    </Cell>
  );
};
export default App;
```
:::

### 自定义尺寸

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';

const App = () => {
  return (
    <>
      <Cell>
        <Progress percent={30} strokeWidth="5" />
      </Cell>
      <Cell>
        <Progress percent={50} strokeWidth="10" />
      </Cell>
      <Cell>
        <Progress percent={70} strokeWidth="15" />
      </Cell>
    </>
  );
};
export default App;
```
:::

### 设置状态显示

:::demo
```jsx
import  React from "react";
import { Progress, Cell } from '@nutui/nutui-react';
import { Checked, Issue } from '@nutui/icons-react';

const App = () => {
  return (
    <>
      <Cell>
        <Progress
          percent={30}
          color="linear-gradient(270deg, rgba(18,126,255,1) 0%,rgba(32,147,255,1) 32.815625%,rgba(13,242,204,1) 100%)"
          animated
        />
      </Cell>
      <Cell align="center">
        <Progress percent={100} />
        <Checked color="green" style={{ margin: '0 5px' }} />
      </Cell>
      <Cell align="center">
        <Progress
          percent={100}
          color="linear-gradient(90deg, rgba(180,236,81,1) 0%,rgba(66,147,33,1) 100%)"
          strokeWidth="15"
        />
        <Issue color="red" style={{ margin: '0 5px' }} />
      </Cell>
    </>
  );
};
export default App;
```
:::

### 动态改变

:::demo
```jsx
import  React, {useState} from "react";
import { Progress, Cell, Button } from '@nutui/nutui-react';

const App = () => {
  const [value, setValue] = useState(0);
  return (
    <>
      <Cell align="center">
        <Progress percent={value} />
        <span style={{ margin: '0 5px' }}>{value}%</span>
      </Cell>
      <Cell>
        <Button
          type="default"
          style={{ margin: 8 }} 
          onClick={() => {
            let num = value;
            if (value <= 0) {
              return false;
            }
            num -= 10;
            setValue(num);
          }}
        >
          减少
        </Button>
        <Button 
          type="primary" 
          onClick={() => {
            let num = value;
            if (value >= 100) {
              return false;
            }
            num += 10;
            setValue(num);
          }}
        >
          增加
        </Button>
      </Cell>
    </>
  );
};
export default App;
```
:::

## Progress

### Props

| 字段 | 说明 | 类型 | 默认值
|----- | ----- | ----- | -----
| percent | 百分比 | `number` | `0`
| background | 进度条背景颜色 | `string` | `#f3f3f3`
| color | 进度条线条颜色 | `string` | `linear-gradient(135deg, #fa2c19 0%, #fa6419 100%)`
| strokeWidth | 进度条宽度 | `string` | -
| showText | 是否显示进度条文字内容 | `boolean` | `false`
| animated | 进度条当前状态，`true`展示动画效果 | `boolean` | `false`

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称 | 默认值 |
| --- | --- |
| --nutui-progress-height | `10px` |
| --nutui-progress-border-radius | `12px` |
| --nutui-progress-text-color | `$primary-text-color` |
| --nutui-progress-text-padding | `0 5px` |
| --nutui-progress-text-font-size | `9px` |
| --nutui-progress-text-position-top | `-4px`|
| --nutui-progress-text-position-bottom | `-4px`|
| --nutui-progress-text-border-radius | `5px` |
