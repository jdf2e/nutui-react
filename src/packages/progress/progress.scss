.nut-progress {
  display: flex;
  align-items: center;
  position: relative;
  width: 100%;

  &-outer {
    flex: auto;
    border-radius: $progress-border-radius;
    height: $progress-height;
    background: $progress-background;

    .nut-progress-active:before {
      content: '';
      position: absolute;
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
      border-radius: $progress-border-radius;
      animation: progressActive 2s ease-in-out infinite;
    }
  }

  &-inner {
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    transition: all 0.4s;
    border-radius: $progress-border-radius;
    background: $progress-color;
  }

  &-text {
    display: flex;
    align-items: center;
    position: absolute;
    top: $progress-text-position-top;
    bottom: $progress-text-position-bottom;
    transition: all 0.4s;
    transform: translate(-50%);

    &-inner {
      display: flex;
      align-items: center;
      height: 100%;
      width: 100%;
      color: $progress-text-color;
      padding: $progress-text-padding;
      border-radius: $progress-text-border-radius;
      font-size: $progress-text-font-size;
      line-height: 1;
      background: $progress-text-background;
    }
  }

  @keyframes progressActive {
    0% {
      background: rgba(255, 255, 255, 0.1);
      width: 0;
    }

    20% {
      background: rgba(255, 255, 255, 0.5);
      width: 0;
    }

    to {
      background: rgba(255, 255, 255, 0);
      width: 100%;
    }
  }
}
