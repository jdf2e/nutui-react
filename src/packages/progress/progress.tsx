import React, { FunctionComponent, ReactNode } from 'react'
import classNames from 'classnames'
import { Checked } from '@nutui/icons-react'
import { BasicComponent, ComponentDefaults } from '@/utils/typings'

export type TextType = 'icon' | 'text'

export interface ProgressProps extends BasicComponent {
  percent: number
  fillColor: string
  color: string
  strokeWidth: string
  textColor: string
  showText: boolean
  textInside: boolean
  textBackground: string
  textType: TextType
  status: boolean
  icon: ReactNode
  children: ReactNode
}

const defaultProps = {
  ...ComponentDefaults,
  percent: 0,
  fillColor: '#f3f3f3',
  color: 'linear-gradient(135deg, #fa2c19 0%, #fa6419 100%)',
  strokeWidth: '',
  textColor: '',
  showText: true,
  textInside: false,
  textBackground: 'linear-gradient(135deg, #fa2c19 0%, #fa6419 100%)',
  textType: 'text',
  status: false,
  icon: null,
  children: undefined,
} as ProgressProps

export const Progress: FunctionComponent<
  Partial<ProgressProps> & React.HTMLAttributes<HTMLDivElement>
> = (props) => {
  const {
    className,
    style,
    percent,
    fillColor,
    color,
    strokeWidth,
    textColor,
    showText,
    textInside,
    textBackground,
    textType,
    status,
    icon,
    children,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  }

  const classPrefix = 'nut-progress'

  const classes = classPrefix

  const classesOuter = classNames({
    [`${classPrefix}-outer`]: true,
    [`${classPrefix}-base`]: true,
  })

  const classesInner = classNames({
    [`${classPrefix}-inner`]: true,
    [`${classPrefix}-active`]: status,
  })

  const classesText = `${classPrefix}-text`

  const classesInsideText = classNames({
    [`${classPrefix}-text`]: true,
    [`${classPrefix}-insidetext`]: true,
  })

  const classesTextInner = classNames({
    [`${classPrefix}-text__inner`]: true,
  })

  const stylesOuter: React.CSSProperties = {
    height: `${strokeWidth}px`,
    background: `${fillColor}`,
  }

  const stylesInner: React.CSSProperties = {
    width: `${percent}%`,
    background: `${color}`,
  }

  const stylesInsideText: React.CSSProperties = {
    left: `${percent}%`,
    background: textBackground || color,
  }

  const stylesInsideIcon: React.CSSProperties = {
    left: `${percent}%`,
  }

  return (
    <div className={`${classes} ${className}`} style={style} {...rest}>
      <div className={classesOuter} style={stylesOuter}>
        <div className={classesInner} style={stylesInner}>
          {showText && textInside && (
            <>
              {children ? (
                <div className={classesInsideText} style={stylesInsideIcon}>
                  {children}
                </div>
              ) : (
                <div className={classesInsideText} style={stylesInsideText}>
                  <span
                    className={classesTextInner}
                    style={{ color: textColor }}
                  >
                    {percent}%
                  </span>
                </div>
              )}
            </>
          )}
        </div>
      </div>
      {showText && !textInside && (
        <div className={classesText}>
          {textType === 'text' && (
            <span className={classesTextInner} style={{ color: textColor }}>
              {percent}%
            </span>
          )}
          {textType === 'icon' && (
            <>{icon || <Checked width={16} height={16} color="#439422" />}</>
          )}
        </div>
      )}
    </div>
  )
}

Progress.defaultProps = defaultProps
Progress.displayName = 'NutProgress'
