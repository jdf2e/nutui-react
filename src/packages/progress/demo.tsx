import React, { useState } from 'react'
import { useTranslate } from '../../sites/assets/locale'
import { Progress } from './progress'
import { Cell } from '@/packages/cell/cell'
import { Icon } from '@/packages/icon/icon'
import { Button } from '@/packages/button/button'
import Toast from '../toast'

interface T {
  basic: string
  customStyle: string
  noShowPercentage: string
  showPercentage: string
  showInsidePercentage: string
  customContent: string
  customSize: string
  statusDisplay: string
  dynamicChange: string
  reduce: string
  add: string
}

const ProgressDemo = () => {
  const cellStyles = {
    paddingRight: '30px',
  }
  const [translated] = useTranslate<T>({
    'zh-CN': {
      basic: '基础用法',
      customStyle: '线形进度条-设置颜色高度',
      noShowPercentage: '百分比不显示',
      showPercentage: '百分比外显',
      showInsidePercentage: '百分比内显',
      customContent: '百分比内显自定义',
      customSize: '自定义尺寸',
      statusDisplay: '状态显示',
      dynamicChange: '动态改变',
      reduce: '减少',
      add: '增加',
    },
    'zh-TW': {
      basic: '基礎用法',
      customStyle: '線形進度條-設置顏色高度',
      noShowPercentage: '百分比不顯示',
      showPercentage: '百分比外顯',
      showInsidePercentage: '百分比內顯',
      customContent: '百分比內顯自定義',
      customSize: '自定義尺寸',
      statusDisplay: '狀態顯示',
      dynamicChange: '動態改變',
      reduce: '減少',
      add: '增加',
    },
    'en-US': {
      basic: 'Basic Usage',
      customStyle: 'Custom Style',
      noShowPercentage: 'Don’t Show Percentage',
      showPercentage: 'Percentage displayed outside',
      showInsidePercentage: 'Percentage displayed inside',
      customContent: 'Custom Content',
      customSize: 'Custom Size',
      statusDisplay: 'Status Display',
      dynamicChange: 'Dynamic Change',
      reduce: 'reduce',
      add: 'add',
    },
  })

  const [value, setValue] = useState(0)

  return (
    <>
      <div className="demo">
        <h2>{translated.basic}</h2>
        <Cell style={cellStyles}>
          <Progress percentage={30} />
        </Cell>
        <h2>{translated.customStyle}</h2>
        <Cell style={cellStyles}>
          <Progress
            percentage={30}
            strokeColor="rgba(250,44,25,0.47)"
            stroke-width="20"
            textColor="red"
          />
        </Cell>
        <h2>{translated.noShowPercentage}</h2>
        <Cell style={cellStyles}>
          <Progress percentage={50} showText={false} />
        </Cell>
        <h2>{translated.showPercentage}</h2>
        <Cell style={cellStyles}>
          <Progress percentage={30} />
        </Cell>
        <h2>{translated.showInsidePercentage}</h2>
        <Cell style={cellStyles}>
          <Progress percentage={60} textInside />
        </Cell>
        <h2>{translated.customContent}</h2>
        <Cell style={cellStyles}>
          <Progress percentage={60} textInside>
            <Icon
              size={30}
              name="https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png"
            />
          </Progress>
        </Cell>
        <h2>{translated.customSize}</h2>
        <Cell style={cellStyles}>
          <Progress percentage={30} size="small" textInside />
        </Cell>
        <Cell style={cellStyles}>
          <Progress percentage={50} size="base" textInside />
        </Cell>
        <Cell style={cellStyles}>
          <Progress percentage={70} size="large" textInside />
        </Cell>
        <h2>{translated.statusDisplay}</h2>
        <Cell style={cellStyles}>
          <Progress
            percentage={30}
            strokeColor="linear-gradient(270deg, rgba(18,126,255,1) 0%,rgba(32,147,255,1) 32.815625%,rgba(13,242,204,1) 100%)"
            status
          />
        </Cell>
        <Cell style={cellStyles}>
          <Progress percentage={100} textType="icon" />
        </Cell>
        <Cell style={cellStyles}>
          <Progress
            percentage={100}
            strokeColor="linear-gradient(90deg, rgba(180,236,81,1) 0%,rgba(66,147,33,1) 100%)"
            strokeWidth="15"
            textType="icon"
            iconName="issue"
            iconColor="red"
          />
        </Cell>
        <h2>{translated.dynamicChange}</h2>
        <Cell style={cellStyles}>
          <Progress percentage={value} />
        </Cell>
        <Cell style={cellStyles}>
          <Button
            type="default"
            style={{ margin: 8 }}
            // eslint-disable-next-line consistent-return
            onClick={() => {
              let num = value
              if (value <= 0) {
                Toast.text('进度已为0')
                return false
              }
              num -= 10
              setValue(num)
            }}
          >
            {translated.reduce}
          </Button>
          <Button
            type="primary"
            style={{ margin: 8 }}
            // eslint-disable-next-line consistent-return
            onClick={() => {
              let num = value
              if (value >= 100) {
                Toast.text('进度已为100%')
                return false
              }
              num += 10
              setValue(num)
            }}
          >
            {translated.add}
          </Button>
        </Cell>
      </div>
    </>
  )
}

export default ProgressDemo
