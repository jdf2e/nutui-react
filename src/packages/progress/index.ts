import { Progress } from './progress'

export type { ProgressProps } from './progress'
export default Progress
