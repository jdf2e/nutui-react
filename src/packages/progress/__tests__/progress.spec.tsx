import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Progress } from '../progress'

test('should render progress when use props', async () => {
  const { asFragment, container } = render(<Progress percentage={100} />)
  expect(asFragment()).toMatchSnapshot()
})
test('should render different height and color when use color height props', async () => {
  const { container } = render(
    <Progress percentage={50} strokeColor="blue" strokeWidth="20" textColor="red" />
  )
  let inner = container.querySelector('.nut-progress-inner')
  expect(inner?.getAttribute('style')).toBe('width: 50%; border-radius: 12px; background: blue;')
  let span = container.querySelector('.nut-progress-text span')
  expect(span?.getAttribute('style')).toBe('color: red;')
})

test('should hide percentage when use showText props', () => {
  const { container } = render(<Progress percentage={30} showText={false} />)
  let text = container.querySelector('.nut-progress-text')
  expect(text).toBeNull()
})

test('should render inside percentage when use textInside props', () => {
  const { container } = render(<Progress percentage={50} textInside={true} />)
  let text = container.querySelector('.nut-progress-text')
  expect(text).toHaveClass('nut-progress-text nut-progress-insidetext')
})

test('should render custom size when use size props', () => {
  const { container } = render(<Progress percentage={50} size="large" />)
  expect(container.querySelector('.nut-progress-outer')).toHaveClass('nut-progress-large')
})
