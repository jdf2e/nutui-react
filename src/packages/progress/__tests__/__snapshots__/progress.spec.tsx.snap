// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render progress when use props 1`] = `
<DocumentFragment>
  <div
    class="nut-progress "
    iconclassprefix="nut-icon"
    iconfontclassname="nutui-iconfont"
  >
    <div
      class="nut-progress-outer nut-progress-base"
      style="border-radius: 12px; background: rgb(243, 243, 243);"
    >
      <div
        class="nut-progress-inner"
        style="width: 100%; border-radius: 12px;"
      />
    </div>
    <div
      class="nut-progress-text"
    >
      <span
        class="nut-progress-text__inner"
        style="color: rgb(51, 51, 51);"
      >
        100%
      </span>
    </div>
  </div>
</DocumentFragment>
`;
